solr/core/src/java/org/apache/solr/handler/admin/api/ReplaceNodeAPI.java

```diff
@@ -0,0 +1,121 @@
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.parameters.RequestBody;
+import java.util.HashMap;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import org.apache.solr.client.solrj.SolrResponse;
+import org.apache.solr.common.cloud.ZkNodeProps;
+import org.apache.solr.common.params.CollectionParams;
+import org.apache.solr.common.params.CollectionParams.CollectionAction;
+import org.apache.solr.core.CoreContainer;
+import org.apache.solr.handler.admin.CollectionsHandler;
+import org.apache.solr.jersey.JacksonReflectMapWriter;
+import org.apache.solr.jersey.PermissionName;
+import org.apache.solr.jersey.SolrJerseyResponse;
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.response.SolrQueryResponse;
+
+/**
+ * V2 API for recreating replicas in one node (the source) on another node(s) (the target).
+ *
+ * <p>This API is analogous to the v1 /admin/collections?action=REPLACENODE command.
+ */
+@Path("cluster/nodes/{sourceNodeName}/commands/replace/")
+public class ReplaceNodeAPI extends AdminAPIBase {
+
+  @Inject
+  public ReplaceNodeAPI(
+      CoreContainer coreContainer,
+      SolrQueryRequest solrQueryRequest,
+      SolrQueryResponse solrQueryResponse) {
+    super(coreContainer, solrQueryRequest, solrQueryResponse);
+  }
+
+  @POST
+  @Produces({"application/json", "application/xml", BINARY_CONTENT_TYPE_V2})
+  @PermissionName(COLL_EDIT_PERM)
+  public SolrJerseyResponse replaceNode(
+      @Parameter(description = "The name of the node to be replaced.", required = true)
+          @PathParam("sourceNodeName")
+          String nodeName,
+      @RequestBody(description = "Contains user provided parameters", required = true)
+          ReplaceNodeRequestBody requestBody)
+      throws Exception {
+    final SolrJerseyResponse response = instantiateJerseyResponse(SolrJerseyResponse.class);
+    final CoreContainer coreContainer = fetchAndValidateZooKeeperAwareCoreContainer();
+    /** TODO Record node for log and tracing */
+    final ZkNodeProps remoteMessage = createRemoteMessage(nodeName, requestBody);
+    final SolrResponse remoteResponse =
+        CollectionsHandler.submitCollectionApiCommand(
+            coreContainer,
+            coreContainer.getDistributedCollectionCommandRunner(),
+            remoteMessage,
+            CollectionParams.CollectionAction.REPLACENODE,
+            DEFAULT_COLLECTION_OP_TIMEOUT);
+    if (remoteResponse.getException() != null) {
+      throw remoteResponse.getException();
+    }
+
+    disableResponseCaching();
+    return response;
+  }
+
+  public ZkNodeProps createRemoteMessage(String nodeName, ReplaceNodeRequestBody requestBody) {
+    final Map<String, Object> remoteMessage = new HashMap<>();
+    remoteMessage.put(SOURCE_NODE, nodeName);
+    remoteMessage.put(TARGET_NODE, requestBody.targetNodeName);
```

Comment:
[0] If the user doesn't provide a request-body, e.g.

```
curl -ilk -X POST "http://localhost:8983/api/cluster/nodes/localhost:7574_solr/commands/replace"
```

then this line will cause a NullPointerException.

I'm a little torn on this - I think it'd technically be "correct" if we forced users to provide an empty JSON object (`{}`) as the request body even if they don't want to specify any of the optional parameters.  But in practice - idk, it seems like it'd be a pain that we could save our users.

What do you think of surrounding these lines with a `if (requestBody != null)` check?

Also, this would be a great this to have a test for later on when you get to that point.